### PR TITLE
Changes to CasSys and VersionInfo

### PR DESCRIPTION
**Pull request by vahbuna**
Changes to format of output file
Display build date in YYYY-MM-DD format

### CODE REVIEW

**Comment by martinec** [comment]
@vahbuna  Would you mind to create two pull requests ? One for each issue: `Update standoff output` and `Add display build date time`. Thank you.

File: VersionInfo.cpp [hunk 1]
```diff
@@ -181,15 +185,176 @@ static void fill_compiler_info(unichar* s) {
 
 
 #ifdef __DATE__
+// see https://stackoverflow.com/questions/11697820/how-to-use-date-and-time-predefined-macros-in-as-two-integers-then-stri
```
Comment (martinec) [inline VersionInfo.cpp]: Use instead the macro: `UNITEX_VERSION_BUILD_DATE` already defined on [Version.h](UnitexGramLab/unitex-core/blob/master/Version.h). FYK, the value is assigned at building time as `date '+%B %d, %Y'` by [Vinber](UnitexGramLab/vinber-backend/blob/master/vinber.sh#L46). Other related macros are:

`UNITEX_VERSION_BUILD_DAY `
`UNITEX_VERSION_BUILD_MONTH `
`UNITEX_VERSION_BUILD_YEAR`
`UNITEX_VERSION_BUILD_TIMESTAMP `

Comment (vahbuna) [inline VersionInfo.cpp]: But how can I test this on the local machine

Comment (martinec) [inline VersionInfo.cpp]: What about this:

1. Use only `UNITEX_VERSION_BUILD_DATE` on `VersionInfo.cpp` no extra logic is needed

The, you have several alternatives, for exemple:

2a. Customize your `Makefile` to redefine `UNITEX_VERSION_BUILD_DATE` as explained here: https://stackoverflow.com/a/31336873/204287, or

2b. Create a `VersionLocal.h` header redefining `UNITEX_VERSION_BUILD_DATE`

```cpp
// process __DATE__ here
#undef UNITEX_VERSION_BUILD_DATE
#define UNITEX_VERSION_BUILD_DATE BUILD_YEAR_CH0 BUILD_YEAR_CH1 ...
```
3. Then include conditionally `VersionLocal.h` on `VersionInfo.h`

```cpp
...
#include "Version.h"
...
#if (defined(UNITEX_VERSION_BUILD_IS_ANONYMOUS) &&\
             UNITEX_VERSION_BUILD_IS_ANONYMOUS)
# include "VersionLocal.h"
#endif
...
```

**Comment by gvollant** [comment]
J’ai essayé de créer un revert

 

Il faudrait modifier le code pour votre suggestion avec les .H. Il faut que cela reste compatible avec les makefile qui ne précisent pas explicitement la date

 

De : Cristian Martinez [mailto:notifications@github.com] 
Envoyé : vendredi 30 juin 2017 22:19
À : UnitexGramLab/unitex-core
Cc : gvollant; State change
Objet : Re: [UnitexGramLab/unitex-core] Changes to CasSys and VersionInfo (#32)

 

@martinec commented on this pull request.

  _____  

In VersionInfo.cpp <https://github.com/UnitexGramLab/unitex-core/pull/32#discussion_r125122619> :

> @@ -181,15 +185,176 @@ static void fill_compiler_info(unichar* s) {
 
 
 #ifdef __DATE__
+// see https://stackoverflow.com/questions/11697820/how-to-use-date-and-time-predefined-macros-in-as-two-integers-then-stri

What about this:

1.	Use only UNITEX_VERSION_BUILD_DATE on VersionInfo.cpp no extra logic is needed

The, you have several alternatives, for exemple:

2a. Customize your Makefile to redefine UNITEX_VERSION_BUILD_DATE as explained here: https://stackoverflow.com/a/31336873/204287, or

2b. Create a VersionLocal.h header redefining UNITEX_VERSION_BUILD_DATE

// process __DATE__ here
#undef UNITEX_VERSION_BUILD_DATE
#define UNITEX_VERSION_BUILD_DATE BUILD_YEAR_CH0 BUILD_YEAR_CH1 ...

3.	Then include conditionally VersionLocal.h on VersionInfo.h

...
#include "Version.h"
...
#if (defined(UNITEX_VERSION_BUILD_IS_ANONYMOUS) &&\
             UNITEX_VERSION_BUILD_IS_ANONYMOUS)
# include "VersionLocal.h"
#endif
...

**Pull request merged**